app/models/approval.rb

```diff
@@ -66,10 +59,15 @@ def approved_at
     end
   end
 
+  # Used by the state machine
+  def on_rejected_entry(new_state, event)
+    self.cart.update_approval_status
+    Dispatcher.on_approval_status_change(self)  # todo - move this out
+  end
 
-  private
-
-  def set_default_status
-    self.status ||= 'pending'
+  # Used by the state machine
+  def on_approved_entry(new_state, event)
+    self.cart.update_approval_status
+    Dispatcher.on_approval_status_change(self)  # todo - move this out
   end
```

Comment:
I would normally say we should pull the contents of these two identical methods into a common function, but I don't think they *should* be the same, or that they will stay the same for very long. Adding the `ThreeStateWorkflow` to the `Cart` will probably take care of this.